tweet.py

```diff
@@ -19,6 +21,16 @@
 access_token_secret = environ['access_token_secret']
 
 
+def get_img_random():
+    no=0
+    while(True):
+        no = random.randint(0,99)
+        if(os.path.exists('post_'+str(no)+'.png')):
+            return 'post_'+str(no)+'.png'
+        else:
+            continue

```

Comment:
This is a nice approach. Trying to decide what's the best approach to error out when there's no image file just so it doesn't get stuck in an infinite loop... 

Could we keep track of the number of times it enters the else clause so that when it reaches a threshold (whatever we want), it breaks out of the while loop ? Do you think that's feasible ?

Reply:
Indeed. The two approaches we can take are -
1) Count the no. of times we enter the else clause and break the loop when a threshold has been reached. (As you mentioned above)
2) A linear search for files numbered 1 to 100 via os.path.exists and a flag variable.

The latter approach seems more reasonable to me due to the assurance that we'll either have no file or some file but it's also more resource taxing, in case we increase the no of images used in future.

Reply:
I agree with you on the latter approach! It gives more assurance, whereas in my solution, we run the risk of not giving it enough time before finding the image! However, it can take quite an amount of resource if we do have more images in the future.

Regarding this concern, could we: 

1. Just extract all the file names from the image folder into an array in the .py file (chances are we'd have to loop through and then it won't be much different than a linear search) 
2. We perform whatever search to find the name based on the random function we already have
3. Or we can just do `random.choice(img_array)` to randomly do it from there. 

I think this is less resource taxing but could be slower if we have to fetch all the names of images every time we call the function.

Reply:
Made a linear scan of appropriate filenames into an array and a `random.choice(img_array)` to select and post an image.

Reply:
Awesome! I'll take a look soon. Thank you thank you

Reply:
I looked at the image folders and saw that the file name conventions have 'a' after some of them (16, 16a, etc). Could you incorporate this in the condition while searching for file names, when you have time ?

Reply:
Done. 
Not sure what the difference between the two types of files is though yet.
Awaiting some explanation in #90 .

Reply:
Thank you so much !

Reply:
There's nothing different between the files with different naming conventions. I just failed to keep the same convention